File: src/com/jda/functionalprograms/PowerOfTwo.java
/**************************************************
 * Purpose: Finds the nth power of 2
 *
 * @author Mohan Pednekar
 * @version 1.0
 * @since 05-07-2018
 **************************************************/

package com.jda.functionalprograms;

import com.jda.utility.Maths;
import com.jda.utility.Printer;
import com.jda.utility.Reader;
import com.jda.utility.StringUtils;
import com.jda.utility.Validation;

public
class PowerOfTwo {

  public static
  void main(String[] args) {

    Reader reader = new Reader();

    // checks whether a command line argument is provided
    if (!Validation.checkCmdArgs(args, 1)) { return; }

    int n = StringUtils.stringToPositiveInteger(args[0]);
    if (n == -1) { return; }
    if (n >= 31) { Printer.printLine("Please input a power less than 31"); }
    int result = Maths.power(2, n);

    Printer.printLine("The " + StringUtils.ordinal(n) + " power of 2 is " + result);
  }
}

File: src/com/jda/functionalprograms/PowerOfTwo.kt
/**************************************************
 * Purpose: Finds the nth power of 2
 *
 * @author Mohan Pednekar
 * @version 1.0
 * @since 05-07-2018
 */

package com.jda.functionalprograms

import com.jda.utility.Utility

object PowerOfTwo {

    @JvmStatic
    fun main(args: Array<String>) {

        val reader = Utility().Reader()

        // checks whether a command line argument is provided
        if (!Utility.checkCmdArgs(args, 1)) {
            return
        }

        val n = Utility.stringToPositiveInteger(args[0])
        if (n == -1) {
            return
        }
        if (n >= 31) {
            Utility.printLine("Please input a power less than 31")
        }
        val result = Utility.power(2, n)

        Utility.printLine("The " + Utility.ordinal(n) + " power of 2 is " + result)
    }
}
